icicle/tests/CMakeLists.txt

```diff
@@ -0,0 +1,48 @@
+cmake_minimum_required(VERSION 3.18)
+
+include(../cmake/Common.cmake)
+include(../cmake/CurvesCommon.cmake)
+include(../cmake/FieldsCommon.cmake)
+
+set_env()
+set_gpu_env()
+
+project(icicle_benchmarks LANGUAGES CUDA CXX)
+
+if(DEFINED CURVE)
+  if(DEFINED FIELD)
+    message( FATAL_ERROR "CURVE and FIELD cannot be defined at the same time" )
+  endif ()
+  check_curve()
+elseif(DEFINED FIELD)
+  check_field()
+endif ()
+
+set(CMAKE_CUDA_FLAGS "${CMAKE_CUDA_FLAGS} -DG2=ON")
```

Comment:
Maybe it's worth moving G2 to an option here?

Reply:
Yes that's very fair. Curve tests have been written before grumpkin was added and they just assume that every curve has G2 and import names like `g2_affine_t` and `g2_projective_t` without conditional compilation. So it was just easier to enable G2 in cmake, but really I should make G2 conditional in tests, will do that